auth/src/main/java/com/firebase/ui/auth/data/remote/FacebookSignInHandler.java

```diff
@@ -0,0 +1,153 @@
+package com.firebase.ui.auth.data.remote;
```

Comment:
TODO(me): add tests

Reply:
I think just for review sanity you should do the tests in another PR.  This one is already 1k+ lines and tests will probably double it.

Reply:
SGTM for the individual provider handlers since they work AFAIK, but I've actually found a bunch of issues with my `ProvidersHandler` implementation thanks to some lovely TDD (so I'll just add that one for now).